### PR TITLE
Run Checkstyle, RevApi, and Spotbugs linting using verify goal

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

This PR resolves an issue where Checkstyle, RevApi, and Spotbugs linting may miss some source code files (specifically in samples) as validation was running using the specific goals for each linting tool. The problem there was that running the full Maven build may include additional source and test sources, such as samples. A change is being made here to run the linting steps as part of the full Maven build rather than having a two-stage approach where the project was built then another Maven run was performed to run the linting goals.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/pipelines/code-quality-reports.yml [hunk 1]
```diff
@@ -87,34 +87,18 @@ extends:
...
-                  options: '$(DefaultOptions) -T 2C -DskipTests -Dgpg.skip -DtrimStackTrace=false -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dcheckstyle.skip -Dspotbugs.skip -Drevapi.skip=true -Djacoco.skip=true'
-                  mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-                  javaHomeOption: 'JDKVersion'
-                  jdkVersionOption: $(JavaBuildVersion)
-                  jdkArchitectureOption: 'x64'
-                  publishJUnitResults: false
-                condition: and(succeeded(), ne(variables['LintingGoals'], ''))
-
               - task: Maven@4
                 displayName: 'Run Linting'
```
Comment (srnagar) [inline eng/pipelines/code-quality-reports.yml:91]: Update the displayName to indicate this does both - build and linting

File: eng/pipelines/templates/steps/run-and-validate-linting.yml [hunk 2]
```diff
@@ -20,20 +20,6 @@ parameters:
...
-      javaHomeOption: 'JDKVersion'
-      jdkVersionOption: ${{ parameters.JavaBuildVersion }}
-      jdkArchitectureOption: 'x64'
-      publishJUnitResults: false
-    continueOnError: ${{ parameters.ContinueOnError }}
-    condition: or(eq(coalesce(variables['IsLatestNonLtsJdk'], true), true), and(${{ parameters.RunLinting }}, succeeded()))
-
   - task: Maven@4
     ${{ if ne(parameters.SDKType, 'client') }}:
       displayName: 'Run SpotBugs and Checkstyle'
```
Comment (srnagar) [inline eng/pipelines/templates/steps/run-and-validate-linting.yml:25]: Same here - fix display name

File: eng/pipelines/templates/steps/run-and-validate-linting.yml [hunk 3]
```diff
@@ -42,12 +28,11 @@ steps:
...
-        goals: 'checkstyle:check spotbugs:check'
+        options: '$(DefaultOptions) --no-transfer-progress -T ${{parameters.BuildParallelization}} -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dspotless.apply.skip=true -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.skip=true -pl $(ProjectList) -am'
       ${{ else }}:
-        options: '$(DefaultOptions) --no-transfer-progress -T 1C -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl $(ProjectList)'
-        goals: 'checkstyle:check spotbugs:check revapi:check'
-      mavenOptions: '$(MemoryOptions)'
+        options: '$(DefaultOptions) --no-transfer-progress -T ${{parameters.BuildParallelization}} -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dspotless.apply.skip=true -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl $(ProjectList) -am'
```
Comment (srnagar) [inline eng/pipelines/templates/steps/run-and-validate-linting.yml:33]: Should we skip javadoc?

**Review by srnagar — APPROVED** [review]

File: eng/pipelines/templates/steps/run-and-validate-linting.yml (hunk 3 above)
Comment (alzimmermsft) [reply eng/pipelines/templates/steps/run-and-validate-linting.yml:33]: For this part we skip Javadoc, there is another Maven task run to run Javadoc later on. This is done as CheckStyle, RevApi, and Spotbugs all have the ability to output a file that contains the issues found but Javadoc only logs to console. So, this one is run in parallel as we don't have a concern for interlaced output then a single thread job is run for Javadoc to ensure the output can be read.

**Pull request merged**